src/effects/native/autopaneffect.h

```diff
@@ -0,0 +1,104 @@
+#ifndef AUTOPANEFFECT_H
+#define AUTOPANEFFECT_H
+
+#include <QMap>
+
+#include "util.h"
+#include "util/defs.h"
+#include "util/types.h"
+#include "engine/effects/engineeffect.h"
+#include "engine/effects/engineeffectparameter.h"
+#include "effects/effectprocessor.h"
+#include "sampleutil.h"
+
+// This class provides a float value that cannot be increased or decreased
+// by more than a given value to avoid clicks.
+// Finally I use it with only one value but i wonder if it can be useful
+// somewhere else (I hear clicks when I change the period of flanger for example).
+class RampedSample {
+  public:
+    
+    inline RampedSample()
+        : ramped(false),
+          maxDifference(1.0f),
+          initialized(false) {}
+    
+    virtual ~RampedSample(){};
+    
+    inline void setRamping(const float newMaxDifference) {
+        maxDifference = newMaxDifference;
+    }
+    
+    inline RampedSample & operator=(const float newValue) {
```

Comment:
This is a nice trick to hide functions from a reviewer ;-)

I would prefer not to override =, instead find a function name that reflects what it does.